components/TextFields/src/Theming/MDCTextInputControllerUnderline+MaterialTheming.m

```diff
@@ -0,0 +1,69 @@
+// Copyright 2020-present the Material Components for iOS authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import "MDCTextInputControllerUnderline+MaterialTheming.h"
+
+@implementation MDCTextInputControllerUnderline (MaterialTheming)
+
+- (void)applyThemeWithScheme:(id<MDCContainerScheming>)scheme {
+  // Color
+  [self applyColorThemeWithColorScheme:scheme.colorScheme];
+
+  // Typography
+  [self applyTypographyThemeWithScheme:scheme.typographyScheme];
+}
+
+- (void)applyColorThemeWithColorScheme:(id<MDCColorScheming>)colorScheme {
+  UIColor *onSurface87Opacity = [colorScheme.onSurfaceColor colorWithAlphaComponent:(CGFloat)0.87];
+  UIColor *onSurface60Opacity = [colorScheme.onSurfaceColor colorWithAlphaComponent:(CGFloat)0.60];
+
+  self.activeColor = colorScheme.primaryColor;
+  self.errorColor = colorScheme.errorColor;
+  self.normalColor = onSurface87Opacity;
+  self.inlinePlaceholderColor = onSurface60Opacity;
+  self.trailingUnderlineLabelTextColor = onSurface60Opacity;
+  self.leadingUnderlineLabelTextColor = onSurface60Opacity;
+
+  if ([self conformsToProtocol:@protocol(MDCTextInputControllerFloatingPlaceholder)]) {
+    id<MDCTextInputControllerFloatingPlaceholder> textInputControllerFloatingPlaceholder =
+        (id<MDCTextInputControllerFloatingPlaceholder>)self;
+
+    if ([textInputControllerFloatingPlaceholder
+            respondsToSelector:@selector(setFloatingPlaceholderNormalColor:)]) {
+      textInputControllerFloatingPlaceholder.floatingPlaceholderNormalColor =
```

Comment:
In the original Themer, the code was

```obcj
UIColor *primary87Opacity = [colorScheme.primaryColor colorWithAlphaComponent:(CGFloat)0.87];	
textInputControllerFloatingPlaceholder.floatingPlaceholderNormalColor = onSurface60Opacity;	
textInputControllerFloatingPlaceholder.floatingPlaceholderActiveColor = primary87Opacity;
```

Is the floating placeholder now always `primaryColor` rather than onSurface @ 60% or primary @ 87%?  

I think it's fine if the new theming is correct, but it might be worth calling-out in the PR summary so in the future we don't accidentally think it was an oversight.  Is that OK?

Reply:
Ah I think I just missed this I've updated it to match the original.